### PR TITLE
SYS-3943 new batch query endpoint

### PR DESCRIPTION
**Pull request by nahuseyoum**

### CODE REVIEW

File: lib/apis/query.ts [hunk 1]
```diff
@@ -79,11 +97,7 @@ export class Query {
       throw new Error('Invalid server response');
     }
 
-    if (response.data.result) {
```
Comment (nahuseyoum) [inline lib/apis/query.ts:82]: @fluorostani can you think of a reason why this would cause issues?

Comment (fluorostani) [reply lib/apis/query.ts:82]: Looking at the gateway-side, we return response.data.error.code and response.data.error.message, if you wanted to use them?

Comment (fluorostani) [reply lib/apis/query.ts:82]: From https://github.com/Aventus-Network-Services/avn-gateway-api/blob/4b4cf94e75ecd0d18d71ebfc11e1aaf5b5b48003/avn-gateway/common/utils.js#L74-L80

Comment (fluorostani) [reply lib/apis/query.ts:82]: Otherwise, if you're just ignoring those error types then it should work as is

**Review by fluorostani — APPROVED** [review]

File: lib/apis/query.ts (hunk 1 above)
Comment (nahuseyoum) [reply lib/apis/query.ts:82]: I think we expect the connector to return a non 200 status code (like a 5xx or 4xx)  if there is an error. Those fields will be useful for the library running this sdk. This change was to allow a `null` or `undefined` as a legitimate successful response. For example when searching for NFT by ID, a null means the nft is not found. Before this change, for some reason, the sdk always expected a non null value.

Comment (fluorostani) [reply lib/apis/query.ts:82]: Ah, gotcha. All good then, sorry

**Pull request merged**